### DISCUSSION TITLE
Comments extension

### DISCUSSION THREAD

**Discussion by kobrynovych** [ueberdosis/tiptap#5272] [General Questions]
Hi,
I integrate comments in a tiptap text editor using this code

https://tiptap.dev/docs/editor/comments/overview

I understand that these comments are stored on your server, I wonder if the data about the comments will not disappear over time

**1.
how long will they be stored on your server? will it happen that I have the signature 'Entry' and after 14 days the written comment will be deleted on your server?**

**2.
if the comments will be deleted after some time on your server and even if we clone the comments to our server through a web hook, how can these cloned comments be combined with those that are still on your server? 
(I would be grateful for an example of the code or a link to the documentation on this)**

my vue2 code:

               ```

                  import { Editor, EditorContent, BubbleMenu } from '@tiptap/vue-2';
                  import StarterKit from '@tiptap/starter-kit';
                  import * as Y from 'yjs'
                  import Collaboration from '@tiptap/extension-collaboration'
                  import { TiptapCollabProvider } from '@hocuspocus/provider'
                  import { CommentsKit } from '@tiptap-pro/extension-comments'  
                 
  
                  data() {
                      return {
                          editor: null,
                          ydoc: null,
                          provider: null,
                          threads: [],
                     }
                 },


             initEditor() {
                this.ydoc = new Y.Doc()

                this.provider = new TiptapCollabProvider({
                    name: this.name,
                    appId: this.appId,
                    token: this.token,
                    document: this.ydoc,
                })

                this.provider.on('synced', this.onProviderSynced)
                this.provider.watchThreads(this.onUpdateThreads)       

                this.editor = new Editor({
                    // content: this.editorContent,        // Note: for Collaboration not use
                    extensions: [
                        StarterKit.configure({
                            history: false,               
                        }),
                        Collaboration.configure({
                            document: this.ydoc,
                        }),
                        CommentsKit.configure({                    
                            provider: this.provider,
                        }),
                    ],
                })
             },
                


                 onProviderSynced(e) {
                     if (!this.ydoc?.getMap('config')?.get('initialContentLoaded') && this.editor) {
                         this.ydoc?.getMap('config')?.set('initialContentLoaded', true)
                         this.editor.commands.setContent(this.editorContentProps)  
                     }

                     this.onUpdateThreads()   
                },



                onUpdateThreads() {
                    this.threads = this.provider.getThreads()
               },

```

thank you very much for your time on this question, I will be very grateful for an answer and code examples of how to combine the cloned data with the ones on your server

**Comment by janthurau** [comment]
hey @kobrynovych, your data will not be deleted while you have an active subscription, so no need to worry about it. The comments are actually just saved in the ydoc, so if you have the ydoc on your own database, you also have the comments.

You can fetch the ydoc using the management API.

**Comment by kobrynovych** [comment]
Hey  @janthurau , Thank you very much for your response.

It turns out that our backend project is built on PHP/Laravel, and it seems we cannot store the YJS format.
https://tiptap.dev/docs/hocuspocus/server/examples#php--laravel

It seems that in this case, we just need to store the document data in JSON or HTML format in our database, and the comment data (to keep it for a rainy day), but not send them to the frontend in props, since we are already using data from Tiptap servers.

This is how we get document data:


```

this.ydoc = new Y.Doc()

this.provider = new TiptapCollabProvider({
    name: this.name,
    appId: this.appId,
    token: this.token,
    document: this.ydoc,
})

```



And this is how we get comment data:



`
this.threads = this.provider.getThreads()

`

Am I saying this correctly?
